### PR TITLE
BUG: DTA.isin with mismatched resos

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Also trims a bit off of the upcoming DTA._from_sequence non-nano PR

### CODE REVIEW

File: pandas/tests/series/methods/test_isin.py [hunk 1]
```diff
@@ -43,6 +43,26 @@ def test_isin_with_string_scalar(self):
...
+        tm.assert_series_equal(result, expected)
+
+        s_values = day_values.astype("M8[s]")
+        dta = type(ser._values)._simple_new(s_values, dtype=s_values.dtype)
+        result = ser.isin(dta)
+        tm.assert_series_equal(result, expected)
+
+        # FIXME(2.0): DTA._from_sequence incorrectly treats Timestamp[s].value
+        #  as nanoseconds.
+        # result = ser.isin(list(dta))
```
Comment (mroeschke) [inline pandas/tests/series/methods/test_isin.py:63]: xfail this case?

Comment (jbrockmendel) [reply pandas/tests/series/methods/test_isin.py:63]: good idea, will update

Comment (jbrockmendel) [reply pandas/tests/series/methods/test_isin.py]: updated + green

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**